action/hook/table.go

```diff
@@ -0,0 +1,96 @@
+// Copyright (c) 2020 Target Brands, Inc. All rights reserved.
+//
+// Use of this source code is governed by the LICENSE file in this repository.
+
+package hook
+
+import (
+	"sort"
+	"time"
+
+	"github.com/go-vela/cli/internal/output"
+
+	"github.com/go-vela/types/library"
+
+	"github.com/dustin/go-humanize"
+	"github.com/gosuri/uitable"
+)
+
+// table is a helper function to output the
+// provided hooks in a table format with
+// a specific set of fields displayed.
+func table(hooks *[]library.Hook) error {
+	// create new table
+	table := uitable.New()
+
+	// set column width for table to 50
+	table.MaxColWidth = 50
+
+	// ensure the table is always wrapped
+	table.Wrap = true
+
+	// set of hook fields we display in a table
+	table.AddRow("NUMBER", "STATUS", "EVENT", "BRANCH", "CREATED")
+
+	// iterate through all hooks in the list
+	for _, h := range reverse(*hooks) {
+		// calculate created timestamp in human readable form
+		c := humanize.Time(time.Unix(h.GetCreated(), 0))
+
+		// add a row to the table with the specified values
+		table.AddRow(h.GetNumber(), h.GetStatus(), h.GetEvent(), h.GetBranch(), c)
+	}
+
+	// output the table in default format
+	err := output.Default(table)
+	if err != nil {
+		return err
+	}
+
+	return nil
+}
+
+// wideTable is a helper function to output the
+// provided hooks in a wide table format with
+// a specific set of fields displayed.
+func wideTable(hooks *[]library.Hook) error {
+	// create new wide table
+	table := uitable.New()
+
+	// set column width for wide table to 200
+	table.MaxColWidth = 200
+
+	// ensure the wide table is always wrapped
+	table.Wrap = true
+
+	// set of hook fields we display in a table
+	table.AddRow("NUMBER", "SOURCE", "STATUS", "HOST", "EVENT", "BRANCH", "CREATED")
```

Comment:
In all the other table outputs I've seen you append the wide fields onto the end of the row, any reason why that changed here? And should we update this to append `SOURCE` and `HOST` to be consistent in this case?

Reply:
If the desire is to always append onto the end then I can do so 👍 

But, currently this isn't a pattern established in the code base (even before this refactor effort) so I'd have to go back and update some existing functions.

As an example, I'll use the `go-vela/cli/action/build` pkg:

https://github.com/go-vela/cli/blob/96a1c21a3373ab5d2c008f3b2f6f530e9bef9c51/action/build/table.go#L35

vs

https://github.com/go-vela/cli/blob/96a1c21a3373ab5d2c008f3b2f6f530e9bef9c51/action/build/table.go#L69

If you're wondering why that pkg is also not following the pattern of appending to the end, it's because that was how the original code was setup so I was keeping it the same as was previously written.

https://github.com/go-vela/cli/blob/96a1c21a3373ab5d2c008f3b2f6f530e9bef9c51/cmd/build/get.go#L146

https://github.com/go-vela/cli/blob/96a1c21a3373ab5d2c008f3b2f6f530e9bef9c51/cmd/build/get.go#L127

Reply:
I'm indifferent just wanted to ask the question, I thought I saw a pattern going on but wanted to double-check. I'm fine with just leaving the order since we have so many other changes going into this release.

Reply:
Sure 👍 

I'm good either way!

I think we'll eventually want to create a pattern like you're suggesting anyways..

But, I also agree that a lot of changes are going into the codebase for this release..

Not sure where we should draw that line of too many changes 🤷‍♂️

Reply:
I think the pattern good be a good first issue since it's just proposing a pattern and implementing it. So, I can merge this and we can create that maybe?

Reply:
Sounds good to me 👍